utils/oomparser/oomparser.go

```diff
@@ -0,0 +1,194 @@
+	parsedLine := containerRegexp.FindStringSubmatch(line)
+	if parsedLine == nil {
+		return nil
+	}
+	currentOomInstance.ContainerName = parsedLine[1]
+	return nil
+}
+
+// gets the pid, name, and date from a line and adds it to oomInstance
+func getProcessNamePid(line string, currentOomInstance *OomInstance) (bool, error) {
+	reList := lastLineRegexp.FindStringSubmatch(line)
+	if reList == nil {
+		return false, nil
+	}
+	linetime, err := time.Parse(time.Stamp, reList[1])
+	if err != nil {
+		return false, err
+	}
+	currentOomInstance.TimeOfDeath = linetime
+	if err != nil {
+		return false, err
+	}
+	pid, err := strconv.Atoi(reList[2])
+	if err != nil {
+		return false, err
+	}
+	currentOomInstance.Pid = pid
+	currentOomInstance.ProcessName = reList[3]
+	return true, nil
+}
+
+// uses regex to see if line is the start of a kernel oom log
+func checkIfStartOfOomMessages(line string) (bool, error) {
+	potential_oom_start := firstLineRegexp.MatchString(line)
+	if potential_oom_start {
+		return true, nil
+	}
+	return false, nil
+}
+
+// opens a reader to grab new messages from the Reader object called outPipe
+// opened in PopulateOomInformation.  It reads line by line splitting on
+// the "\n" character.  Checks if line might be start or end of an oom message
+// log. Then the
+// lines are checked against a regexp to check for the pid, process name, etc.
+// At the end of an oom message group, AnalyzeLines adds the new oomInstance to
+// oomLog
+func (self *OomParser) analyzeLines(outPipe io.ReadCloser, outStream chan *OomInstance) {
+	ioreader := bufio.NewReader(outPipe)
+	line, err := ioreader.ReadString('\n')
+	for err == nil {
+		in_oom_kernel_log, err := checkIfStartOfOomMessages(line)
+		if err != nil {
+			glog.Errorf("%v", err)
+			continue
+		}
+		if in_oom_kernel_log {
+			oomCurrentInstance := new(OomInstance)
+			finished := false
+			for err == nil && !finished {
+				err = getContainerName(line, oomCurrentInstance)
+				if err != nil {
+					glog.Errorf("%v", err)
+				}
+				finished, err = getProcessNamePid(line, oomCurrentInstance)
+				if err != nil {
+					glog.Errorf("%v", err)
+				}
+				line, err = ioreader.ReadString('\n')
+			}
+			in_oom_kernel_log = false
+			self.oomInstances = append(self.oomInstances, oomCurrentInstance)
+			outStream <- oomCurrentInstance
+		}
+		line, err = ioreader.ReadString('\n')
+	}
+	glog.Errorf("%v", err)
+}
+
+// looks for system files that contain kernel messages and if one is found, sets
+// the systemFile attribute of the OomParser object
+func getSystemFile() (string, error) {
+	const varLogMessages = "/var/log/messages"
+	const varLogSyslog = "/var/log/syslog"
+	if utils.FileExists(varLogMessages) {
+		return varLogMessages, nil
+	} else if utils.FileExists(varLogSyslog) {
+		return varLogSyslog, nil
+	}
+	return "", errors.New("Neither /var/log/messages nor /var/log/syslog exists from which to read kernel errors")
```

Comment:
Errors.New() won't take %s arguments as fmt will, apparently.  I could construct a string with the variables concatenated or just leave them in the string.